lib/api/src/grpc/proto/collections_internal_service.proto

```diff
@@ -36,3 +41,18 @@ message WaitForShardStateRequest {
   ReplicaState state = 3;  // Shard state to wait for
   uint64 timeout = 4; // Timeout in seconds
 }
+
+message GetShardRecoveryPointRequest {
+  string collection_name = 1; // Name of the collection
+  uint32 shard_id = 2; // Id of the shard
+}
+
+message GetShardRecoveryPointResponse {
+  RecoveryPoint recovery_point = 1; // Recovery point of the shard
+  double time = 2; // Time spent to process
+}
+
+message RecoveryPoint {
+  // TODO: use different type here, that does not have force flag
+  repeated ClockTag clocks = 1;
+}
```

Comment:
I currently reuse the `ClockTag` for this:

```proto
message ClockTag {
  uint64 peer_id = 1;
  uint32 clock_id = 2;
  uint64 clock_tick = 3;
  bool force = 4;
}
```

We need the first three fields, but the `force` flag is obsolete here. I can make it optional, or we create a new type with just the three fields.

What do you think? I prefer not to bloat our interface with a whole bunch of types.

Reply:
I'd add a new type. Like, we won't be able to change it once we release, and I don't want to tie `ClockTag` to the `RecoveryPoint`. What if we want to add more fields to `ClockTag` in the future?

I'd also, maybe, make it a `map<ClockKey, uint64>` or something like this, but I guess it does not matter much, cause we'll probably convert gRPC type to "collection" type anyway.

Reply:
> Like, we won't be able to change it once we release, and I don't want to tie ClockTag to the RecoveryPoint. What if we want to add more fields to ClockTag in the future?

Good point about future maintainability. I'll add a new type.

> I'd also, maybe, make it a map<ClockKey, uint64> or something like this, but I guess it does not matter much, cause we'll probably convert gRPC type to "collection" type anyway.

I tried this at first, but it is not possible. We cannot use custom types as keys, sadly.

It also seems to introduce overhead by sorting the map and things like that.

Reply:
Added in d99708e134c1582376073485e7a8ec06cf915cd1